pkg/kubelet/status/manager.go

```diff
@@ -182,16 +195,42 @@ func (m *manager) TerminatePods(pods []*api.Pod) bool {
 				Terminated: &api.ContainerStateTerminated{},
 			}
 		}
-		select {
-		case m.podStatusChannel <- podStatusSyncRequest{pod, pod.Status}:
-		default:
+		newStatus := m.updateStatusInternal(pod, pod.Status)
+		if newStatus != nil {
+			select {
+			case m.podStatusChannel <- podStatusSyncRequest{pod.UID, *newStatus}:
```

Comment:
I like the idea of letting periodic sync dealing with this, but I am afraid that it may increase the latency -- user deletes a terminated pod has to wait the extra 0~10s for the pod to actually be deleted.